### PR TITLE
[AB] Upgrade deprecated environments based on the provided replacement

### PR DESCRIPTION
**Pull request by janinko**
### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/bacon/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

**Review by jomrazek — COMMENTED** [review]
Are you aware this could potentially break builds?

**Comment by janinko** [comment]
It should not break builds as the upgrade path should be veted by PNC admins to be safe.  But yes.

File: experimental/src/main/java/org/jboss/bacon/experimental/impl/generator/BuildConfigGenerator.java [hunk 1]
```diff
@@ -233,8 +235,26 @@ public BuildConfig copyExisting(BuildConfiguration bc, BuildConfigurationRevisio
         return buildConfig;
     }
 
+    private static void setEnvironment(BuildConfig buildConfig, Environment env, boolean useName) {
+        if (useName) {
+            if (!env.getName().equals(buildConfig.getEnvironmentName())) {
+                log.debug(
```
Comment (jomrazek) [inline experimental/src/main/java/org/jboss/bacon/experimental/impl/generator/BuildConfigGenerator.java:241]: is it worth to change it into info level?

**Review by jomrazek — APPROVED** [review]

File: experimental/src/main/java/org/jboss/bacon/experimental/impl/generator/BuildConfigGenerator.java (hunk 1 above)
Comment (janinko) [reply experimental/src/main/java/org/jboss/bacon/experimental/impl/generator/BuildConfigGenerator.java]: well, Kuba said that it should not bother users, but sure

Comment (jomrazek) [reply experimental/src/main/java/org/jboss/bacon/experimental/impl/generator/BuildConfigGenerator.java]: well I would say it's informing not bothering and if they don't want to be bothered they can update their environment. Thank you

**Pull request merged**